integrations/gitlab-v2/gitlab/clients/gitlab_client.py

```diff
@@ -139,3 +144,39 @@ async def _process_single_group(
                     f"Fetched {len(resource_batch)} {resource_type} for group {group_id}"
                 )
                 yield resource_batch
+
+    async def get_group_members(
+        self, group_id: str, include_bot_members: bool
+    ) -> AsyncIterator[list[dict[str, Any]]]:
+        async for batch in self.rest.get_paginated_group_resource(group_id, "members"):
+            if batch:
+                filtered_batch = batch
+                if not include_bot_members:
+                    filtered_batch = [
+                        member
+                        for member in batch
+                        if "bot" not in member["username"].lower()
+                    ]
+                logger.info(
+                    f"Received batch of {len(filtered_batch)} members for group {group_id}"
+                )
+                yield filtered_batch
+
+    async def enrich_group_with_members(
+        self, group: dict[str, Any], include_bot_members: bool
+    ) -> dict[str, Any]:
+        logger.info(f"Enriching group {group['id']} with members")
+        members = [
+            {
+                "username": member["username"],
+                "name": member["name"],
+                "id": member["id"],
+            }
+            async for members_batch in self.get_group_members(
+                group["id"], include_bot_members
+            )
+            for member in members_batch
+        ]
```

Comment:
I think this will be more readable and easier to debug
```suggestion
members = []
async for members_batch in self.get_group_members(group["id"], include_bot_members):
    for member in members_batch:
        members.append({
            "username": member["username"],
            "name": member["name"],
            "id": member["id"]
        })
```